enhancements/operator-dev-doc.md

```diff
@@ -0,0 +1,269 @@
+
+### How can I test changes to an OpenShift operator/operand/release component? 
+
+When developing on OpenShift, at some point you'll want to test your changes in a running cluster.  Any component image in a release payload
+can be substituted for a locally built image.  You can test your changes in an OpenShift cluster like so:     
+```
+NOTE:
+If using `quay.io`,  repositories are by default private.  You'll need to
+visit `quay.io` and change settings to public for any new image repos you push.
+This is to allow OpenShift to pull your image.
+```
+
+#### OPTION A)
+#### Build a new release payload and launch a cluster with that release
+
+For this example I'll start with the release image     
+`registry.svc.ci.openshift.org/ocp/release:4.2`     
+and test a change to the `github.com/openshift/openshift-apiserver` repository.     
+
+1. Build the image and push it to a registry (quay.io, docker.io)
+*Dockerfile name varies per repository, this example used Dockerfile.rhel, but it may have a different name*
+```
+$ cd local/path/to/openshift-apiserver
+$ buildah bud -f path/to/Dockerfile.rhel -t quay.io/yourname/openshift-apiserver:test .
+$ buildah push quay.io/yourname/openshift-apiserver:test
+```
+    
+2. Assemble a release payload with your test image and push it to a registry    
+Get the name of the image (`openshift-apiserver`) you want to substitute:
+```
+$ oc adm release info registry.svc.ci.openshift.org/ocp/release:4.2
+```
+If the above command fails, visit the steps [here](#what-is-an-openshift-release-image) to add credentials to
+your local `~/.docker/config.json`.  
+
+This command will assemble a release payload incorporating your test image _and_ will push it to the quay.io repository.    
+Be sure to set this repository in quay.io as `public`.    
+```
+$ oc adm release new --from-release registry.svc.ci.openshift.org/ocp/release:4.2 \
+  openshift-apiserver=quay.io/yourname/openshift-apiserver:test \
+  --to-image quay.io/yourname/release:test
+```
+
+If the above command succeeds, move on to Step 3.
+If the above command fails, you need to authenticate against `quay.io`. 
+Add the necessary credentials to your local `~/.docker/config.json`:
+ - Visit `https://try.openshift.com`, `GET STARTED`, login w/ RedHat account if not already,    
+   choose any `Infrastructure Provider`, `Copy Pull Secret` to a local file (or download)    
+ - Add the quay auth from the pull-secret to ~/.docker/config.json    
+   The config file should have the following:
+   
+```console
+$cat ~/.docker/config.json
+{
+  "auths": {
+    "registry.svc.ci.openshift.org": {
+      "auth": "blahblahblah"
+    },
+    "quay.io": {
+      "auth": "blahblahblah"
+    }
+  }
+}
+```
+
+3. Run the installer with the release payload that has your test image
+    
+```
+$ cd path/to/openshift/installer
+$ make build (if you haven't already)
```

Comment:
try the `oc adm release extract --command openshift-install <release image>`

^^ this requires no env substitution `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE`

Reply:
great, thanks! I did not know about this gem!!